backend/src/main/java/pl/edu/pw/elka/etherscan/EthereumAddressValidator.java

```diff
@@ -0,0 +1,57 @@
+package pl.edu.pw.elka.etherscan;
+
+
+
+public class EthereumAddressValidator {
+
+    private final static int correctAddressLength = 42;
+    private final static char correctFirstChar = '0';
+    private final static char correctSecondChar = 'x';
+
+    public void validateAddress(String address){
+        if(!address.equals("")) {       // This if is only temporary to let test "EtherscanConnectorTests-returnsTransactions" pass
+            validateNull(address);
+            validateLength(address);
+            validateFirstChar(address);
+            validateSecondChar(address);
+            String addressWithout0x = address.replaceFirst("0x", "");
+            validateStructure(addressWithout0x);
+            validateChecksum(addressWithout0x);
+        }
+    }
+
+    private void validateNull(String address) {
+        if(address == null) {
+            throw new BadEthereumAddressException("Wrong Ethereum address");
```

Comment:
moze wszystkie z tych wiadomosci powinny mowic bardziej dokladnie co sie stalo - np jak jest za dlugi to napisac ze jest za dlugi a nie enigmatyczne Wrong Ethereum address?